src/components/tasks/card/AssigneeDropdownMenu.tsx

```diff
@@ -0,0 +1,96 @@
+import { FC, useState, useEffect, SetStateAction } from "react";
+import userType from '@/interfaces/user.type';
+import { ALL_USERS } from '@/components/constants/url';
+import fetch from '@/helperFunctions/fetch';
+import { toast, ToastTypes } from '@/helperFunctions/toast';
+import classNames from '@/components/tasks/card/card.module.scss';
+
+type Props = {
+  cardDetails: any
+  updateAssignee: (username: string) => void
+  setisAssigneeDropdownOpen: React.Dispatch<SetStateAction<boolean>>
+}
+
+//https://github.com/Real-Dev-Squad/website-status/issues/303
+
+const AssigneeDropdownMenu: FC<Props> = ({ cardDetails,updateAssignee,setisAssigneeDropdownOpen }) => {
+  const [searchValue, setSearchValue] = useState("");
+  const [filteredList, setFilteredList] = useState([]);
+  const [assigneeList, setAssigneeList] = useState([])
+  const { ERROR } = ToastTypes;
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const { requestPromise } = fetch({ url: ALL_USERS });
+        const { data } = await requestPromise;
+        const usersData = data.members || data.users
+        setAssigneeList(usersData)
+      } catch (err: any) {
+        toast(ERROR, err.message);
+      }
+    };
+    fetchData();
+
+  }, []);
+
+  useEffect(() => {
+    if (searchValue) {
+      const newFilteredList = assigneeList.filter((member: userType) => (
+        member.username?.toLowerCase().includes(searchValue.toLowerCase())
+      ));
+      setFilteredList(newFilteredList)
```

Comment:
cant this go directly into the onChangeHandler rather than setting the state first and then running a useffect to change a state again.

Reply:
yeah makes sense, i don't remember what i was thinking at that time while writing this

Reply:
No problem, this is the reason we have PR review 😇